tests/basic/clocks.rs

```diff
@@ -437,3 +440,37 @@ fn clock_fetch_update() {
         None,
     );
 }
+
+fn clock_once(num_threads: usize) {
+    let once = Arc::new(Once::new());
+    let init = Arc::new(std::sync::atomic::AtomicUsize::new(0));
+
+    let threads = (0..num_threads)
+        .map(|_| {
+            let once = Arc::clone(&once);
+            let init = Arc::clone(&init);
+            thread::spawn(move || {
+                check_clock(|i, c| (c > 0) == (i == 0));
+                once.call_once(|| init.store(me(), std::sync::atomic::Ordering::SeqCst));
+                let who_inited = init.load(std::sync::atomic::Ordering::SeqCst);
+                // should have inhaled the clock of the thread that inited the Once, but might also
+                // have inhaled the clocks of threads that we were racing with for initialization
```

Comment:
I don't understand the second part of this comment.  We are asserting here that this thread's clock is nonzero (causally depends on) only for (1) the thread that inited the Once, (2) itself, (3) the main thread.
Maybe just remove the second part of the comment?

Reply:
We're asserting that the clock is non-zero for *at least* those three threads; we're saying nothing about the others. The comment is just to say why the check isn't stronger -- we might have learned something about threads other than those three depending on whether they raced for initialization or not.

Reply:
Ah, sorry.  For some reason I was thinking of an equivalence; but it's just an implication.